source/Octopus.Tentacle/Scripts/RunningScript.cs

```diff
@@ -53,37 +67,51 @@ public void Execute()
                             token,
                             log))
                         {
-                            RunScript(shellPath, writer);
+                            State = ProcessState.Running;
+
+                            if (stateRepository != null)
```

Comment:
This entire if block could perhaps be re-factored out to a method `RecordScriptHasStarted()`